src/main.ts

```diff
@@ -1,16 +1,135 @@
+      core.debug(`resource: ${JSON.stringify(resourceChange)}`)
+      const actions = resourceChange.change.actions
+      const resourceName = `${resourceChange.type} - ${resourceChange.name}`
+      if (actions.length === 1 && actions.includes(Action.create)) {
+        toCreate.push(resourceName)
+      } else if (actions.length === 1 && actions.includes(Action.delete)) {
+        toDelete.push(resourceName)
+      } else if (
+        actions.length === 2 &&
+        actions.includes(Action.delete) &&
+        actions.includes(Action.create)
+      ) {
+        toReplace.push(resourceName)
+      } else if (actions.length === 1 && actions.includes(Action.update)) {
+        toUpdate.push(resourceName)
+      } else {
+        core.debug(`Not found? ${actions}`)
+      }
+    }
+
+    core.debug(`toCreate: ${toCreate}`)
+    core.debug(`toUpdate: ${toUpdate}`)
+    core.debug(`toReplace: ${toReplace}`)
+    core.debug(`toDelete: ${toDelete}`)
 
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    let body = `${commentPrefix}\n`
+    if (toCreate.length > 0) {
+      body += `will create ${toCreate.length} resources: \n`
+      for (const resource of toCreate) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (toUpdate.length > 0) {
+      body += `will update ${toUpdate.length} resources: \n`
+      for (const resource of toUpdate) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (toReplace.length > 0) {
+      body += `will replace (**delete** then create) ${toReplace.length} resources: \n`
+      for (const resource of toReplace) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (toDelete.length > 0) {
+      body += `will **delete** ${toDelete.length} resources: \n`
+      for (const resource of toDelete) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (
+      toCreate.length === 0 &&
+      toUpdate.length === 0 &&
+      toReplace.length === 0 &&
+      toDelete.length === 0
+    ) {
+      body += 'No changes'
+    } else {
+      // TODO find a way to link directly to job/step
+      body += `[see details](https://github.com/${owner}/${repo}/actions/runs/${process.env['GITHUB_RUN_ID']})`
+    }
 
-    core.setOutput('time', new Date().toTimeString())
+    // find previous comment if it exists
+    const comments = await octokit.issues.listComments({
+      owner,
+      repo,
+      issue_number: pr.number
+    })
+    let previousCommentId: number | null = null
+    for (const comment of comments.data) {
+      if (
+        comment.user.login === 'github-actions[bot]' &&
+        comment.body.startsWith(commentPrefix)
+      ) {
+        previousCommentId = comment.id
+      }
+    }
+    if (previousCommentId) {
+      // update the previous comment
+      core.debug(`Updating existing comment ${previousCommentId}`)
```

Comment:
It'd be nice if this was logged even without debug mode. It'd be even more helpful if it had a link to the comment.

Reply:
I haven't looked at the actual function, but the API that `octokit.issues.updateComment` uses [provides a URL to the comment](https://developer.github.com/v3/issues/comments/#response-4) ([as does `createComment`](https://developer.github.com/v3/issues/comments/#response-3)).

Reply:
Done